### ISSUE TITLE
piller render error lacking details

### ISSUE THREAD

**Issue by steverweber** [saltstack/salt#21271]
I think I remember seeing an issue that the pillar render error was hidden because it 'could' expose some passwords to the CLI.

I feel displaying the detailed error is more helpful and important as the default behaviour in our organization's enviroment.

Should have a config for extra details.
```
[root@salt salt-sls]# salt oat-db-\*dev.salt\* state.highstate test=True
oat-db-1-dev.salt.math.xxxxxx.ca:
    Data failed to compile:
----------
    Pillar failed to render with the following messages:
----------
    Rendering SLS 'mfcf.service.oat2.db' failed. Please see master log for details.

```

**Comment by basepi** [comment]
The issue is that the message shown above is returned from the minion. The minion is not allowed to see the pillar errors, because it could contain templating data which would give that minion information it shouldn't have. So if we added the actual error here, then a user using salt-call on the minion could potentially see data they shouldn't.

It's possible that we may be able to inject this data master-side to keep it secure. But I'm leaning towards just relying on the master log for the details:

> Please see master log for details.

**Comment by steverweber** [comment]
I figured the user friendly solution would be running the following:
```
salt --log-level=critical \* state.highstate test=True
```
Well... that wont work but I learned some more salt fundamental.

I think adding a master config var like 'hide_pillar_detailed_error' and default to True is a good solution.

My consern is our salt master system is going to have a few cooks in the kitchen, the log will have extra action in it making it kinda busy.

Also im toying with the idea of locking some users out from viewing the master.log because it could have sensitive data in it.  These locked down users will have limited access to some files including their pillar.  They will need some way to debug a bad pillar without bugging me :)
